src/ngMessages/messages.js

```diff
@@ -44,25 +50,34 @@ var jqLite = angular.element;
  * </form>
  * ```
  *
- * Now whatever key/value entries are present within the provided object (in this case `$error`) then
- * the ngMessages directive will render the inner first ngMessage directive (depending if the key values
- * match the attribute value present on each ngMessage directive). In other words, if your errors
- * object contains the following data:
+ * In order to show error messages corresponding to `myField` we first create a container element with an `ngMessages` attribute
+ * set to the `$error` object owned by the `myField` input in our `myForm` form.
+ *
+ * Within this container we create separate elements for each of the possible errors that `myField` could be annotated with. Exactly
+ * which error a particular element corresponds to is declared using the `ngMessage` attribute - for example, `ng-message="required"`
+ * specified that this particular element should be displayed when the corresponding field is `required` but has no value.
+ *
+ * By default, `ngMessages` will only display one error message for a particular field at any one time. Whether or not an error (child)
+ * element will be displayed is determined by whether the corresponding property in the error object is `true`, and the position of
+ * that element relative to other relevant error elements in the container - error elements which appear first in the DOM will be
+ * displayed above others in cases where more than one such element is deemed to be relevant.
+ *
+ * Given the following error object for our example (which informs us that the field `myField` is currently erroneous in terms of both its
+ * `required` and its `minlength` atrributes):
  *
  * ```javascript
  * <!-- keep in mind that ngModel automatically sets these error flags -->
  * myField.$error = { minlength : true, required : true };
  * ```
+ * The `required` message will be displayed first to the user since it appears before the `minlength` message in the DOM.
```

Comment:
This is actually a bad example ( and has been before your changes), because the minlength error is not set when nothing is entered into input. email + minlength is probably a better idea.